tools/azsdk-cli/docs/specs/6-package-updater.spec.md

```diff
@@ -0,0 +1,669 @@
+- **Metadata and Documentation Update**: Consolidated set of actions to ensure non-source assets (README, changelog, CI config, version numbers, additional descriptors) are accurate and consistent.
+- **Tool**: A discrete CLI (and MCP agent-exposed) command implementing one cohesive post-generation preparation task.
+- **Inline Script Command**: A single shell command provided directly in config that overrides a script path.
+
+---
+
+## Background / Problem Statement
+
+### Current State
+
+After code generation, build, tests, and sample validation, engineers must perform repetitive cross-language housekeeping: updating changelogs, synchronizing README and language-specific files, adjusting package versions, exporting API surfaces (.NET), and inserting or validating CI configuration. Today these tasks are:
+
+- Partially automated in ad-hoc scripts per language / repo.
+- Inconsistent between management-plane and data-plane SDKs (especially version bump timing and changelog handling).
+- Error-prone when performed manually (missed version increments, stale README examples, inconsistent metadata presented across different files, like `pom.xml`, `_metadata.json`).
+- A barrier for integrating an AI agent workflow because required parameters and step order are not encapsulated as composable tools.
+
+### Why This Matters
+
+Without a unified tool layer:
+
+- Contributors spend time re-learning differing per-language conventions.
+- CI or release pipelines fail late due to missing metadata or mismatched versions.
+- Agents cannot reliably assist because state transitions ("now bump version", "now update changelog") aren't formalized.
+- Management-plane vs data-plane discrepancies introduce confusion (e.g., when to bump versions or generate changelog entries).
+
+Formalizing Stage 6 as cohesive tools improves reliability, supports automation-first workflows, and prepares for future LLM-assisted tasks.
+
+---
+
+## Goals and Exceptions/Limitations
+
+### Goals
+
+- [ ] Provide four composable, idempotent tools exposed via CLI and MCP agent: CI Update, Changelog Update, Version Update, Metadata Update.
+- [ ] Normalize execution order and expected outputs across all supported languages.
+- [ ] Encode management-plane vs data-plane behavioral differences explicitly (especially version & changelog logic).
+- [ ] Emit structured JSON results (status + message + next steps hints) enabling orchestration by AI agents and pipelines.
+- [ ] Support configuration-driven per-language scripting (path or inline) without recompiling the tool.
+
+### Exceptions
+
+| Exception | Description | Impact | Workaround |
+|---|---|---|---|
+| Data‑Plane Deferred Version Bump | Data‑plane packages typically defer version adjustment and release‑date updates until the release‑preparedness phase. | The `update-version` tool may be a no‑op for data‑plane flows during Stage 6. | Keep version changes manually for milestone 1. |
```

Comment:
Do we have any ideas for how we will handle this after scenario 1? 

My thought is that the data-plane logic will be put in the same tooling, and it'll be up to our instructions to let the agent know when this tool should be called for data-plane vs mgmt-plane.

Reply:
The logic to determine the SDK type is within the tool code since the tool is expected to run in multiple scenarios: agent, CI pipeline, and local CLI.